pkg/simulator/simulator.go

```diff
@@ -89,6 +91,36 @@ func Run(ctx context.Context, list *gcs.ActionList, cfg *model.SimConfig) (*mode
 		}
 	}
 
+	// perform our final iteration with logger enabled
+	handler.InitLogger()
+	var err error
+	go func() {
+		select {
+		case <-ctx.Done():
+			err = errors.New("error: ctx is done")
+			return
+		case err = <-errChan:
+			return
+		case result := <-resp:
+			for _, a := range aggregators {
+				a.Add(result)
+			}
+		}
+	}()
+
+	err2 := pool.QueueJob(workerpool.Job{
+		Script: list,
+		Config: proto.Clone(cfg).(*model.SimConfig),
+	})
+	if err != nil {
+		// something went wrong in thread
+		return nil, err
+	}
+	if err2 != nil {
+		//context must have been cancelled
+		return nil, err
+	}
+
```

Comment:
I would just delete all simulator changes here. This package should just be deleted and functionality merged into `cmd/srsim`, just haven't had the time to do it.

The reality is we do not want the log run to happen unless explicitly requested. In the gcsim ui (refactor) we do not bundle the logs with the results because it would bloat size and instead just provide a seed selector + generate button which will run the log generation (IE: https://simimpact.app/sh/tgrbhNNBPJcJ#tab=sample).

Instead, should add a new `Run` function in `simulation` to be able to run a single iteration with logging enabled. That's what we do in gcsim, where directly in the `cli` code we just do "if asked to generate logs, make the call to generate": https://github.com/genshinsim/gcsim/blob/feature-stats-rework/cmd/gcsim/main.go#L145-L166

For devs + UI development, we'll want the cli to have a "serve" mode where it spins up an http server and makes the logs available so you can just load the UI and view the logs rather than having to parse the json directly. I can do a separate PR to set that up (read: copy/paste from gcsim) and then have cli run the log iteration from there